File: practice/src/main/java/com/decade/practice/web/converters/ChatIdentifierConverter.java
package com.decade.practice.web.converters;

import com.decade.practice.model.domain.embeddable.ChatIdentifier;
import org.springframework.core.convert.converter.Converter;

import java.util.NoSuchElementException;
import java.util.UUID;

public class ChatIdentifierConverter implements Converter<String, ChatIdentifier> {

      @Override
      public ChatIdentifier convert(String source) {
            return extractChatIdentifier(source);
      }

      public static ChatIdentifier extractChatIdentifier(String source) {
            String[] parts = source.split("\\+");
            if (parts.length != 2) {
                  throw new NoSuchElementException();
            }

            UUID first = UUID.fromString(parts[0].trim());
            UUID second = UUID.fromString(parts[1].trim());

            return new ChatIdentifier(first, second);
      }
}

File: practice/src/main/java/com/decade/practice/controllers/converters/ChatIdentifierConverter.kt
package com.decade.practice.controllers.converters

import com.decade.practice.model.domain.embeddable.ChatIdentifier
import org.springframework.core.convert.converter.Converter
import java.util.*

fun String.extractChatIdentifier(): ChatIdentifier {
      val parts = split("+")
      if (parts.size != 2)
            throw NoSuchElementException()
      val (first, second) = parts.map { UUID.fromString(it.trim()) }
      if (first > second)
            throw NoSuchElementException()
      return ChatIdentifier(first, second)
}

class ChatIdentifierConverter : Converter<String, ChatIdentifier> {
      override fun convert(source: String): ChatIdentifier = source.extractChatIdentifier()
}